torch/csrc/distributed/c10d/FlightRecorderDetail.hpp

```diff
@@ -178,32 +187,57 @@ std::vector<typename FlightRecorder<EventType>::Entry> FlightRecorder<
     update_state(r);
     r.start_ = r.end_ = nullptr;
   }
+
+  // Filter out entries from previous epochs
+  // Only keep entries where reset_epoch_ == current_epoch
+  result.erase(
+      std::remove_if(
+          result.begin(),
+          result.end(),
+          [current_epoch](const Entry& e) {
+            return e.reset_epoch_ < current_epoch;
+          }),
+      result.end());
+
   return result;
 }
 
 template <typename EventType>
-// Returns the entry with the given id, if it exists. Otherwise, returns
-// std::nullopt.
+// Returns the index in entries_ for the given id and reset_epoch.
+// Caller must hold mutex_lock before calling this method.
+size_t FlightRecorder<EventType>::getIdxFromId(size_t id, size_t reset_epoch)
+    const {
+  // Look up the starting idx for the given reset epoch
+  auto it = reset_epoch_start_idx_.find(reset_epoch);
+  TORCH_CHECK(it != reset_epoch_start_idx_.end());
+  // Calculate idx based on where the epoch started
+  return (it->second + id) % max_entries_;
```

Comment:
I think this is wrong? because you reset id to 0, we should not consider reset_epoch anymore? I thought reset_epoch is only for dump and retire_id. When you try to get entry, you can just directly get that entry using the index?